modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -48,5 +59,67 @@ object ConstructionService {
 
       NonEmptyList.fromList(accountIdentifiers)
     }
+
+    def transactionToOperations(transaction: Transaction): Either[String, NonEmptyList[Operation]] =
+      refineV[Not[Equal[0L]]](transaction.amount.value.value).flatMap { amount =>
+        val operations: Either[String, List[Operation]] = refineV[Not[Equal[0L]]](-1L * transaction.amount.value.value).map { negAmount =>
+          val negativeOperation = Operation(
+            OperationIdentifier(OperationIndex(0L)),
+            none,
+            OperationType.Transfer,
+            none,
+            AccountIdentifier(transaction.destination, none).some,
+            Amount(AmountValue(negAmount), DAG).some
+          )
+
+          val positiveOperation = Operation(
+            OperationIdentifier(OperationIndex(1L)),
+            none,
+            OperationType.Transfer,
+            none,
+            AccountIdentifier(transaction.destination, none).some,
+            Amount(AmountValue(amount), DAG).some
+          )
+
+          List(negativeOperation, positiveOperation)
+        }
+
+        operations.flatMap { operationList =>
+          NonEmptyList.fromList(operationList) match {
+            case Some(value) => value.asRight
+            case None        => "No valid operations found.".asLeft
+          }
+        }
+      }
+
+    def parseSignedTransaction(hex: Hex): F[Either[ConstructionError, ConstructionParse.ServiceReturn]] =
```

Comment:
No need to use `traverse` there.

```scala
    def parseSignedTransaction(hex: Hex): EitherT[F, MalformedTransaction.type, ConstructionParse.Response]] = {
      val response = for {
        signedTransaction <- KryoSerializer[F].deserialize[Signed[Transaction]](hex.toBytes).toEitherT
        ops <- transactionToOperations(signedTransaction).toEitherT
        proofs <- signedTransaction.proofs.toNonEmptyList.traverse(_.id.hex.toPublicKey).attemptT
        ids = proofs.map(_.toAddress).map(AccountIdentifier(_, none))
      } yield ConstructionParse.Response(ops.toList, ids.toList.some)

      response.leftMap(_ => MalformedTransaction)
    }

    def parseUnsignedTransaction(hex: Hex): Either[MalformedTransaction.type, ConstructionParse.Response] =
      KryoSerializer[F]
        .deserialize[Transaction](hex.toBytes)
        .flatMap(transactionToOperations)
        .map(ops => ConstructionParse.Response(ops.toList, none))
        .leftMap(_ => MalformedTransaction)
```

Also for-comprehension is a syntax sugar to `flatMap` so you can avoid the callback hell: https://en.wiktionary.org/wiki/callback_hell by writing your nested code without nesting